### ISSUE TITLE
[Event Request] Report 593 "Intrastat - Make Disk Tax Auth"

### ISSUE THREAD

**Issue by JalmarazMartn** [microsoft/ALAppExtensions#18315]
Dear Team,

Is it possible to add an event to this report In Intrastat Jnl. Line – OnAfterGetRecord:

**OnBeforeWriteTextLine("Intrastat Jnl. Line ");//New Event**

 OutStreamObj.WRITETEXT(

The event procedure will have this signature:

procedure OnBeforeWriteTextLine(var IntrastatJnlLine: Record "Intrastat Jnl. Line")
begin
end;

Thank you very much!!!

**Comment by Stepanenko-Vadim** [comment]
Please let me know which version and country this code is from, I cannot identify where you want the event

**Comment by JalmarazMartn** [comment]
Sorry, my Code was outdated!! In fact in a sanish 20.1.39764.39901 there will be two event calls, this way:

    local procedure WriteGroupTotalsToFile2021(var IntrastatJnlLine: Record "Intrastat Jnl. Line"; CountryCode: Code[3]; CountryOfOriginCode: Code[3]; StatSystem: Code[1])
    begin
**OnBeforeWriteTextLine(IntrastatJnlLine);//New Event**
        IntrastatFileWriter.WriteLine(



    local procedure WriteGroupTotalsToFile2022(var IntrastatJnlLine: Record "Intrastat Jnl. Line"; CountryCode: Code[3]; CountryOfOriginCode: Code[3]; StatSystem: Code[1])
    begin
**OnBeforeWriteTextLine(IntrastatJnlLine );//New Event**
        IntrastatFileWriter.WriteLine(

Thank you very much!!

> Please let me know which version and country this code is from, I cannot identify where you want the event

Sorry, my Code was outdated!! In fact in a sanish 20.1.39764.39901 there will be two event calls, this way:

    local procedure WriteGroupTotalsToFile2021(var IntrastatJnlLine: Record "Intrastat Jnl. Line"; CountryCode: Code[3]; CountryOfOriginCode: Code[3]; StatSystem: Code[1])
    begin
**OnBeforeWriteTextLine(IntrastatJnlLine);//New Event**
        IntrastatFileWriter.WriteLine(



    local procedure WriteGroupTotalsToFile2022(var IntrastatJnlLine: Record "Intrastat Jnl. Line"; CountryCode: Code[3]; CountryOfOriginCode: Code[3]; StatSystem: Code[1])
    begin
**OnBeforeWriteTextLine(IntrastatJnlLine );//New Event**
        IntrastatFileWriter.WriteLine(

Thank you very much!!

**Comment by Stepanenko-Vadim** [comment]
Seems like you want to alter some values in the line and it would be preferable to be done at the start of procedure, not in the middle.
Will it work for you if it will not be directly before the writeline, but in the start of both procedures?

**Comment by JalmarazMartn** [comment]
Yes, the goal is substitute the Tarif No. before write the line. Due a sectoral legal requirement  we must do this replacement in some items. We can not set this tariff no. in the item because is only for intrastat and we don´t know where tariff no. is used apart in the app.

If you have a better design in the table for this purpose that gives us this feature, do it your own way.

Thanks you for your great attention and professionalism.
